sqld/src/database/write_proxy/replication.rs

```diff
@@ -251,12 +253,12 @@ impl WalIndexMeta {
             .context("invalid generation id from primary")?
             .as_u128();
 
-        if hello_db_id == self.database_id {
+        if hello_db_id != self.database_id {
```

Comment:
It seems that you fixed it here @MarinPostma

Reply:
yes, I was using this to artificially trigger the error while testing